library/src/options.cxx

```diff
@@ -1,4 +1,10 @@
+#include "macros.h"
+
+F3D_SILENT_WARNING_PUSH()
+F3D_SILENT_WARNING_DECL(4996, "deprecated-declarations")
 #include "options.h"
```

Comment:
not sure to follow why we need to silent include  ?

Reply:
Interestingly, the warning is raised on GCC only. The problem is `options.cxx` is implementing the constructors of the struct, which initialize the deprecated field. I would expect that constructors should be guarded, but we need to guard the header include for some reason.

Reply:
that would be my expectation too. As long as it doesnt impact libf3d users im fine with it